lib/syskit/test/instance_requirement_planning_handler.rb

```diff
@@ -83,7 +87,7 @@ def finished?
                     stub_network.announce_replacements(mapped_tasks)
                     stub_network.remove_obsolete_tasks(mapped_tasks)
                 end
-                @planning_tasks.all?(&:finished?)
+                @planning_tasks.all? { |t| t.resolution_success? || t.failed? }
```

Comment:
I would use `t.finished?` instead of `t.failed?`. It's more general and would cover all cases (even though I'm ... pretty sure ... that the only way to quit is to fail.